web/react/components/modal.jsx

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2015 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+export default class Modal extends ReactBootstrap.Modal {
```

Comment:
The thing is that this encourages bad design. When we pass props to a component the component should reflect those props and ultimately the state of the stores. When we react to the modal opening we break this one way flow. 

Every case where you currently use onShow really shouldn't be using it. It only uses it because of bad design of the overall flow. Therefore you should duplicate the logic everywhere and not encourage bad design by building it into the component. This the reason why the react-boostrap component intentionally does not have this function built in.